modelator/Model.py

```diff
@@ -184,7 +185,7 @@ def _check_sample_thread_worker(
 
     def check(
         self,
-        invariants: List[str] = [],
+        invariants: Optional[List[str]] = None,
```

Comment:
Is the Optional type really needed here? Put it another way, is the case `invariants = []` different than `invariants is None` that we would need to treat them differently?

Reply:
Isn't this only a type hint that is equivalent to saying "List[str] or None"? 
And the reason to remove empty list as default argument is that it can cause problems down the line (as described e.g., [here](https://nikos7am.com/posts/mutable-default-arguments/)). In the existing code there were no problems because the list was only assigned, not appended to, but this was a small change for future-proving